### DISCUSSION TITLE
Spotify Integration

### DISCUSSION THREAD

**Discussion by OrbilightSmartHomes** [small-cactus/M.I.L.E.S#8] [General]
If I'm not a Spotify user, how can I use Miles without him constantly trying to connect Spotify?

When I started up, it seemed I couldn't begin working with MILES without first putting in an API key for Spotify. I faked one just to get through to the app, but now it tries to run Spotify constantly. I've tried asking him not to, I told him to disable Spotify Integration.

We'd be interested in collaborating with you soon to get Miles working in luxury homes across South Florida, Nashville, and California.

**Comment by small-cactus** [comment]
Wow! Thank you for your interest in Miles. To answer your question, I just tested his functionality without a real Spotify API key, and it seemed to work fine:

<img width="921" alt="image" src="https://github.com/small-cactus/M.I.L.E.S/assets/125771841/e44a40e0-51f8-407e-bc7f-d9186de75c84">

If you're talking about when Miles opens the Spotify page to ask for permission? That should only happen the first time you talk to him but if you entered a fake API key, I can see why it might keep popping up, to disable that, the easiest way would be to create a burner Spotify account and setup a developer app using the given instructions, you don't need Spotify premium to enable the support, you just need Spotify premium to enable playback and searching songs. Once you have the client ID and client secret, it can be entered into any instance of Miles anywhere on earth and the bug won't happen.

To directly change the Spotify API keys after setup, you can open the apikey.py file and directly paste it, or instead you can set any of the entries to "empty" to trigger setup again. Like this:

```python
spotify_client_secret="empty"
```

If this answer isn't helpful, I can provide you with a specific file just for you that doesn't have Spotify support.
